src/main/drivers/system.c

```diff
@@ -146,14 +146,24 @@ uint32_t micros(void)
     return (ms * 1000) + (usTicks * 1000 - cycle_cnt) / usTicks;
 }
 
-inline uint32_t getCycleCounter(void)
+int32_t getCycleCounter(void)
```

Comment:
The hardware timer is unsigned (`uint32_t`).  Also, since this should just be register access it should always be inlined, as originally created.

Reply:
Not sure why you think that there is any reason to distrust the compiler when it comes to making the right call on inlining or not.

Reply:
It should always be inlined, even on a debug build.